src/System.Slices/System/Buffers/PinnedMemoryHandle.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    public unsafe struct PinnedMemoryHandle<T>
```

Comment:
Ok. I am surprised to see that we are trying to make interop safe - that is lost cause because of it is inherently unsafe. But we gave up on safety for regular code (for now) where it would actually matter...